machine/models/Sender.py

```diff
@@ -0,0 +1,178 @@
+import torch
+import torch.nn as nn
+from torch.nn import functional as F
+from torch.distributions.categorical import Categorical
+
+from .baseRNN import BaseRNN
+from machine.util.gumbel import gumbel_softmax
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+class Sender(BaseRNN):
+	"""
+		Applies a rnn cell to an input sequence and uses Gumbel softmax sampling to
+		generate a output sequence.
+
+		Args:
+			vocab_size (int): size of the vocabulary
+			output_len (int): the length of the sequence to be generated
+			embedding_size (int): the size of the embedding of input variables
+			hidden_size (int): the size of the hidden dimension of the rnn
+			sos_id (int): index of the start of sequence symbol
+        	eos_id (int): index of the end of sequence symbol
```

Comment:
comment indentation is off.
btw make sure you use autopep or pep8 formatting to style your code.